scaper/core.py

```diff
@@ -1738,40 +1742,50 @@ def _generate_audio(self, audio_path, ann, reverb=None,
                             tmpfiles_internal.append(
                                 tempfile.NamedTemporaryFile(
                                     suffix='.wav', delete=False))
-                            # synthesize concatenated/trimmed background
-                            cmb.build(
-                                [e.value['source_file']] * ntiles,
-                                tmpfiles_internal[-1].name, 'concatenate')
+                            # read in background off disk
+                            event_audio_array, event_sample_rate = soundfile.read(
+                                e.value['source_file'], always_2d=True)
+                            # tile the background along the appropriate dimensions
+                            event_audio_array = np.tile(event_audio_array, (ntiles, 1))
+                            event_audio_array = tfm.build_array(
+                                input_array=event_audio_array,
+                                sample_rate_in=event_sample_rate
+                            )
+                            # Write event_audio_array to disk so we can compute LUFS using ffmpeg
+                            soundfile.write(
+                                tmpfiles_internal[-1].name, event_audio_array.T, self.sr)
                             # NOW compute LUFS
                             bg_lufs = get_integrated_lufs(
                                 tmpfiles_internal[-1].name)
 
                             # Normalize background to reference DB.
                             gain = self.ref_db - bg_lufs
+                            gain = self.ref_db + e.value['snr'] - bg_lufs
+                            event_audio_array = np.exp(gain * np.log(10) / 20) * event_audio_array
```

Comment:
is this numerically safe? (or, is this the safest way to achieve this?)

Reply:
This is how it's done in the `sox` source code, so I went with it.